paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -108,6 +108,14 @@ void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
      */
     List<String> listTables(String databaseName) throws DatabaseNotExistException;
 
+    /**
+     * Check if a table exists in this catalog.
+     *
+     * @param identifier Path of the table
+     * @return true if the given table exists in the catalog false otherwise
+     */
+    boolean tableExists(Identifier identifier);
```

Comment:
If we just have a syntactic sugar:
```
  default boolean tableExists(Identifier ident) {
    try {
      return loadTable(ident) != null;
    } catch (TableNotExistException e) {
      return false;
    }
  }
```
OK, that is fine.

Reply:
Thanks, done.